### PR TITLE
[RFC] Fix of bug when callback of dictwatcheradd didn't execute for changedtick

### PR DESCRIPTION
**Pull request by demiurg337**
My first try of contribution to this project (might I did something wrong). It is fix for issue
https://github.com/neovim/neovim/issues/6635

### CODE REVIEW

**Comment by justinmk** [comment]
@demiurg337 Seems like a reasonable direction. Next step would be to add a test case to `test/functional/ex_cmds/dict_notifications_spec.lua`. 

Also if you don't mind, please run `make lint` and fix any errors.

**Comment by demiurg337** [comment]
> @demiurg337 Seems like a reasonable direction. Next step would be to add a test case to `test/functional/ex_cmds/dict_notifications_spec.lua`.
> 
> Also if you don't mind, please run `make lint` and fix any errors.

File: test/functional/ex_cmds/dict_notifications_spec.lua [hunk 1]
```diff
@@ -338,4 +339,25 @@ describe('dictionary change notifications', function()
...
+          function! OnTickChanged(dict, key, value)
+              call rpcnotify(g:channel, 'SendChangeTick', a:key, a:value)
+          endfunction
+          call dictwatcheradd(b:, "changedtick", "OnTickChanged")
+        ]])
+
+        insert("t");
+
+        local res = next_msg()
+        assert.is_true(res[2] == "SendChangeTick")
```
Comment (justinmk) [inline test/functional/ex_cmds/dict_notifications_spec.lua]: can use `helpers.eq()` to compare the values, which gives more informative failures.

Also better to compare the full `next_msg()` result, not just `[2]`.

I made these changes in the merge-commit.

**Comment by justinmk** [comment]
Merged. Thanks @demiurg337 !

File: test/functional/ex_cmds/dict_notifications_spec.lua (hunk 1 above)
Comment (demiurg337) [reply test/functional/ex_cmds/dict_notifications_spec.lua]: ok, understood

**Pull request closed without merging**